### DISCUSSION TITLE
Optional Custom ID

### DISCUSSION THREAD

**Discussion by mauriceackel** [firecmsco/firecms#218] [Ideas]
HI @fgatti675 I was wondering if it would make sense to change to logic of the `customId` property in an entity schema.

I think it is a common use case that you want to **allow** but not **require** the definition of a custom ID. Hence, I propose to change the type of the property to:

```
customId: 'required' | 'optional' | EnumValue;
```

Then, in the validation you would allow to have an empty ID value if and only if `customId === 'optional'`. Then, if the id field is left empty, the ID will be set automatically as usual. If the value is set, however, it will be used as the entity ID.

What do you think about that?

**Comment by jamiegalbreath** [comment]
I need this to use FIRECMS! Please!

**Reply by mauriceackel** [reply]
I created a PR and this should already be part of the most recent release candidate

**Reply by jamiegalbreath** [reply]
Awesome, will look that up

**Reply by mauriceackel** [reply]
https://github.com/Camberi/firecms/pull/219
